src/sentry/api/base.py

```diff
@@ -202,8 +202,11 @@ def respond(self, context=None, **kwargs):
         return Response(context, **kwargs)
 
     def paginate(
-        self, request, on_results=None, paginator_cls=Paginator, default_per_page=100, **kwargs
+        self, request, on_results=None, paginator=None,
+        paginator_cls=Paginator, default_per_page=100, **kwargs
```

Comment:
Can now pass in a pre-instantiated `paginator`, because I need the tagstore abstraction to decide which kind to use.

I verify the caller doesn't try to pass in a `paginator` along with `kwargs` (used to instantiate a Paginator). The alternative to this is refactor all callers of `self.paginate(...)` to pass in a paginator.